drivers/net/phy/qsfp-mem-dfl.c

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: GPL-2.0
+
+/* Intel(R) Memory based QSFP driver For DFL based devices.
+ *
+ * Copyright (C) 2022 Intel Corporation. All rights reserved.
+ */
+#include <linux/dfl.h>
+#include <linux/phy/qsfp-mem.h>
+#include <linux/module.h>
+
+ssize_t dfl_qsfp_connected_show(struct device *dev, struct device_attribute *attr, char *buf)
+{
+	struct qsfp *qsfp = dev_get_drvdata(dev);
+	u32 plugin = qsfp_connected_show(qsfp);
+
+	return sysfs_emit(buf, "%u\n", plugin);
+}
+
+static DEVICE_ATTR_RO(dfl_qsfp_connected);
+
+static struct attribute *qsfp_mem_attrs[] = {
+	&dev_attr_dfl_qsfp_connected.attr,
+	NULL,
+};
+ATTRIBUTE_GROUPS(qsfp_mem);
+
+static int qsfp_dfl_probe(struct dfl_device *dfl_dev)
+{
+	struct device *dev = &dfl_dev->dev;
+	struct qsfp *qsfp;
+	int ret = 0;
+
+	qsfp = devm_kzalloc(dev, sizeof(*qsfp), GFP_KERNEL);
+	if (!qsfp)
+		return -ENOMEM;
+
+	qsfp->base = devm_ioremap_resource(dev, &dfl_dev->mmio_res);
+	if (!qsfp->base)
+		return -ENOMEM;
+
+	qsfp->dev = dev;
+	mutex_init(&qsfp->lock);
+
+	dev_set_drvdata(dev, qsfp);
+
+	ret = qsfp_init_work(qsfp);
+	if (ret != 0) {
+		dev_err(dev, "Failed to initialize delayed work to read QSFP\n");
+		return ret;
+	}
+	return qsfp_register_regmap(qsfp);
+}
+
+static void qsfp_dfl_remove(struct dfl_device *dfl_dev)
+{
+	struct device *dev = &dfl_dev->dev;
+	struct qsfp *qsfp = dev_get_drvdata(dev);
+
+	qsfp_remove_device(qsfp);
+}
+
+#define FME_FEATURE_ID_QSFP 0x13
+
+static const struct dfl_device_id qsfp_ids[] = {
+	{ FME_ID, FME_FEATURE_ID_QSFP },
+	{ }
+};
+
+static struct dfl_driver qsfp_driver = {
+	.drv = {
+		.name = "qsfp-mem",
+		.dev_groups = qsfp_mem_groups,
+	},
+	.id_table = qsfp_ids,
+	.probe = qsfp_dfl_probe,
+	.remove = qsfp_dfl_remove,
+};
+
+module_dfl_driver(qsfp_driver);
+MODULE_DEVICE_TABLE(dfl, qsfp_ids);
```

Comment:
The manual alias definition for backport is missing:
MODULE_ALIAS("dfl:t0000f0013");

Reply:
Added manual alias definition for backport